sdlib/d/gc/tcache.d

```diff
@@ -151,26 +151,52 @@ public:
 	 * See also: https://dlang.org/spec/arrays.html#capacity-reserve
 	 */
 	size_t getCapacity(const void[] slice) {
-		auto pd = maybeGetPageDescriptor(slice.ptr);
-		if (pd.extent is null) {
+		PageDescriptor pd;
+		if (!getAppendablePageDescriptor(slice, pd)) {
 			return 0;
 		}
 
-		// Appendable slabs are not supported.
-		if (pd.isSlab()) {
+		// Slice must not end before valid data ends, or capacity is zero:
+		auto startIndex = slice.ptr - pd.extent.address;
+		auto stopIndex = startIndex + slice.length;
+
+		// If the slice end doesn't match the used capacity, bail.
+		if (stopIndex != pd.extent.usedCapacity) {
 			return 0;
 		}
 
+		return pd.extent.size - startIndex;
+	}
+
+	bool extend(const void[] slice, size_t length) {
+		if (length == 0) {
+			return false;
```

Comment:
Shouldn't this return true? After all,t he operation succeeded: the allocation was extended by 0 bytes.